packages/dendron-cli/src/commands/utils.ts

```diff
@@ -46,26 +63,40 @@ export async function setupEngine(
   });
   let engine: DEngineClient;
   let port: number;
-  let server: any;
+  let server: Server;
   wsRoot = resolvePath(wsRoot, process.cwd());
   if (useLocalEngine) {
     const engine = DendronEngineV2.create({ wsRoot, logger });
     await engine.init();
     return { wsRoot, engine, port: -1, server: createDummyServer() };
   }
-  if (enginePort) {
+  if (enginePort || opts.attach) {
     logger.info({
       ctx: "setupEngine",
       msg: "connecting to engine",
       enginePort,
+      attach: opts.attach,
     });
     const engineConnector = EngineConnector.getOrCreate({
       wsRoot,
     });
     await engineConnector.init({ portOverride: enginePort });
     engine = engineConnector.engine;
-    port = enginePort;
-    server = createDummyServer();
+    if (enginePort) {
+      port = enginePort;
+    } else {
+      // TODO: don't use type assertion
+      port = engineConnector.port!;
+    }
+    if (engineConnector.serverPortWatcher) {
+      // a file watcher is created when engine port is undefined
+      // needs to be cleaned up on server closing
+      server = createDummyServer(async () => {
+        engineConnector.serverPortWatcher?.close();
```

Comment:
since this already has ? at the end of `engineConnector.serverPortWatcher?` - the wrapping if else block should be redundant right?

Reply:
unfortunately, since the call is inside another function, typescript compiler can't infer that